File: frame/src/main/java/com/king/lib/frame/BaseCompactActivity.java
package com.king.lib.frame;

import android.text.TextUtils;

import com.king.lib.frame.annotations.AnnotationConstants;
import com.king.lib.frame.annotations.LoadingFactory;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.DialogFragment;

/**
 * Desc:
 *
 * @author：Jing Yang
 * @date: 2020/7/3 11:17
 */
public class BaseCompactActivity extends AppCompatActivity {

    private DialogFragment loadingDialog;

    public void showProgress() {
        if (loadingDialog == null) {
            loadingDialog = LoadingFactory.createLoading(getLoadingIdentity());
        }
        if (loadingDialog != null) {
            loadingDialog.show(getSupportFragmentManager(), "loadingDialog");
        }
    }

    protected String getLoadingIdentity() {
        return AnnotationConstants.DEFAULT;
    }

    public void dismissProgress() {
        if (loadingDialog != null) {
            loadingDialog.dismissAllowingStateLoss();
        }
    }

}

File: frame_kotlin/src/main/java/com/king/lib/frame_kotlin/BaseCompactActivity.kt
package com.king.lib.frame_kotlin

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

/**
 * Desc:
 * @author：Jing Yang
 * @date: 2020/7/7 13:23
 */
abstract class BaseCompactActivity: AppCompatActivity() {

    companion object {
        val KEY_BUNDLE = "bundle"
    }

    fun showProgress() {

    }

    fun dismissProgress() {

    }

    open fun<AC> startPage(target: Class<AC>) {
        startActivity(Intent().setClass(this, target))
    }

    open fun<AC> startPage(target: Class<AC>, bundle: Bundle) {
        var intent = Intent().setClass(this, target)
        intent.putExtra(KEY_BUNDLE, bundle)
        startActivity(intent)
    }

    open fun<AC> startPageForResult(target: Class<AC>, bundle: Bundle, requestCode: Int) {
        var intent = Intent().setClass(this, target)
        intent.putExtra(KEY_BUNDLE, bundle)
        startActivityForResult(intent, requestCode)
    }

    open fun getIntentBundle(): Bundle? {
        return getIntentBundle(intent)
    }

    open fun getIntentBundle(intent: Intent): Bundle? {
        return intent.getBundleExtra(KEY_BUNDLE)
    }
}